src/apispec/core.py

```diff
@@ -47,6 +49,21 @@ def to_dict(self):
             if v != {}
         }
 
+    def example(self, name, component=None, **kwargs):
+        """Add an example which can be referenced
+
+        :param str name: identifier by which example may be referenced.
+        :param dict component: example fields.
+
+        https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#exampleObject
+        """
+        if name in self._examples:
+            raise DuplicateComponentNameError(
+                'Another example with name "{}" is already registered.'.format(name)
+            )
+        self._examples[name] = component or {}
```

Comment:
Since there is no helper, it makes no sense not to pass component.

This should be 

    self._examples[name] = component

and the signature should be

    def example(self, name, component, **kwargs):

like `security_scheme`.

Reply:
Ah yes!! I forgot about it. Sorry for still being a noob

Reply:
If you want, I can sqaush everything in a single commit again and force push, so that we can group these refactoring under a single commit, else feel free to merge this once the discussion is resolved

Reply:
No pb. Your first shot was pretty good already.

squash + push force would be great.

If you don't, I'll probably do it while merging.

Thanks again.

Reply:
Thanks, I think it looks good